src/utils.c

```diff
@@ -234,3 +234,39 @@ avifBool avifFractionSub(avifFraction a, avifFraction b, avifFraction * result)
     avifFractionSimplify(result);
     return AVIF_TRUE;
 }
+
+avifBool avifToUnsignedFraction(double v, uint32_t * numerator, uint32_t * denominator)
+{
+    if (v < 0 || v > UINT32_MAX) {
+        return AVIF_FALSE;
+    }
+
+    if (round(v) == v) {
+        *numerator = (uint32_t)v;
+        *denominator = 1u;
+        return AVIF_TRUE;
+    }
+
+    if (v < 1.0) {
+        // Maximize precision by having the denominator as large as possible.
```

Comment:
Maybe consider using continued fraction (https://en.wikipedia.org/wiki/Continued_fraction#Best_rational_approximations) to obtain the best rational approximation?